map2loop/data_checks.py

```diff
@@ -0,0 +1,882 @@
+    return (False, "")
+
+
+
+def validate_optional_columns(
+    geodata: geopandas.GeoDataFrame,
+    config: Dict[str, str],
+    optional_columns: List[str],
+    expected_type: Union[Type, Tuple[Type, ...]],
+    check_blank: bool = False,
+    datatype_name: str = "UNKNOWN"
+) -> List[str]:
+
+    warnings = []
+
+    for config_key in optional_columns:
+        column_name = config.get(config_key)
+
+        if not column_name:
+            warning_msg = (
+                f"Configuration key '{config_key}' is missing for datatype '{datatype_name}'. "
+                f"Optional column validation for this key is skipped."
+            )
+            logger.warning(warning_msg)
+            warnings.append(warning_msg)
+            continue  
+
+        if column_name in geodata.columns:
+            # Type Check
+            if not geodata[column_name].apply(lambda x: isinstance(x, expected_type) or pandas.isnull(x)).all():
+                warning_msg = (
+                    f"Datatype {datatype_name.upper()}: Optional column '{column_name}' "
+                    f"(config key: '{config_key}') contains values that are not of type "
+                    f"{expected_type if isinstance(expected_type, type) else expected_type}. "
+                    "Map2loop processing might not work as expected."
+                )
+                logger.warning(warning_msg)
+                warnings.append(warning_msg)
+
+            # Blank String Check (if applicable)
+            if check_blank and issubclass(expected_type, str):
+                if geodata[column_name].str.strip().eq("").any():
+                    warning_msg = (
+                        f"Datatype {datatype_name.upper()}: Optional column '{column_name}' "
+                        f"(config key: '{config_key}') contains blank (empty) string values. "
+                        "Map2loop processing might not work as expected."
+                    )
+                    logger.warning(warning_msg)
+                    warnings.append(warning_msg)
+
+            # Null Value Check
+            if geodata[column_name].isnull().any():
+                warning_msg = (
+                    f"Datatype {datatype_name.upper()}: Optional column '{column_name}' "
+                    f"(config key: '{config_key}') contains NaN or null values. "
+                    "Map2loop processing might not work as expected."
+                )
+                logger.warning(warning_msg)
+                warnings.append(warning_msg)
+                
+        # else:
+        #     warning_msg = (
+        #         f"Datatype {datatype_name.upper()}: Optional column '{column_name}' "
+        #         f"(config key: '{config_key}') is missing from the data. "
+        #     )
+        ####### this might be taking it a bit too far
+        
+            # logger.info(warning_msg)
+            # warnings.append(warning_msg)
```

Comment:
I think keep that as an info log, but not as a warning. Good to have the extra information avaliable if we need it